akka-stream/src/main/scala/akka/stream/scaladsl/Hub.scala

```diff
@@ -424,33 +424,44 @@ private[akka] class BroadcastHub[T](bufferSize: Int) extends GraphStageWithMater
             val startFrom = head
             activeConsumers += 1
             addConsumer(consumer, startFrom)
-            consumer.callback.invoke(Initialize(startFrom))
+            // in case the consumer is already stopped we need to undo registration
+            // note that this will always be invoked for every consumer at some point in time
+            implicit val ec = ExecutionContexts.sameThreadExecutionContext
+            consumer.callback.invokeWithFeedback(Initialize(startFrom)).onFailure {
+              case _: StreamDetachedException ⇒
+                callbackPromise.future.foreach(callback ⇒
+                  callback.invoke(UnRegister(consumer.id, startFrom, startFrom))
```

Comment:
is `callback.invoke` guaranteed to not throw? I think that is important when using sameThreadExecutionContext

Reply:
Tried to figure out but that's a proper labyrinth, so maybe safest to schedule on actual ec